daffodil-core/src/main/scala/edu/illinois/ncsa/daffodil/dsom/RuntimePropertyMixins.scala

```diff
@@ -252,11 +265,20 @@ trait ElementRuntimeValuedPropertiesMixin
     ExpressionCompilers.String.compile(qn, NodeInfo.NonEmptyString, byteOrderRaw)
   }.value
 
-  final lazy val byteOrderEv = LV('byteOrderEv) {
-    val ev = new ByteOrderEv(byteOrderExpr, elementRuntimeData)
-    ev.compile()
-    ev
-  }.value
+  final lazy val byteOrderEv = {
+    byteOrderRaw // must be defined
```

Comment:
Agree. I will fix this so that things do getOrElse, where the Else results in the SDE that is desired, and try to use that idiom for all these things where there is a maybe and a required case. 

Also I'll search and see if the maybe cases are needed. I expect they are, because there are so many situations in DFDL where a property is not required, but lots of code paths that don't want to have to conditionalize for all those cases, so carrying around a Maybe that happens to be Nope avoids code complexity. 

Ex: there is a maybeFillByteEv, because ERDs always contain it, regardless of whether the format actually needs fill byte or not. When a format needs fill byte and it is not defined something elsewhere in the compiler should be asking for the fillByte property and get an SDE. But in rare case where fillByte isn't needed (such as parser-only schema) one might not run into this, in which case the maybeFillByteEv will be Nope.